presto-base-arrow-flight/src/test/java/com/facebook/plugin/arrow/TestingArrowServerUsingDictionaryVector.java

```diff
@@ -0,0 +1,293 @@
+                    ResultSetMetaData metaData = rs.getMetaData();
+                    int columnCount = metaData.getColumnCount();
+
+                    for (int i = 1; i <= columnCount; i++) {
+                        String columnName = metaData.getColumnName(i);
+                        String columnType = metaData.getColumnTypeName(i);
+                        int precision = metaData.getPrecision(i);
+                        int scale = metaData.getScale(i);
+
+                        ArrowType arrowType = convertSqlTypeToArrowType(columnType, precision, scale);
+                        Field field = new Field(columnName, FieldType.nullable(arrowType), null);
+                        fields.add(field);
+                    }
+                }
+            }
+            else {
+                throw new IllegalArgumentException("Either schema_name/table_name or select_statement must be provided.");
+            }
+
+            Schema schema = new Schema(fields);
+            FlightEndpoint endpoint = new FlightEndpoint(new Ticket(flightDescriptor.getCommand()));
+            return new FlightInfo(schema, flightDescriptor, Collections.singletonList(endpoint), -1, -1);
+        }
+        catch (Exception e) {
+            logger.error(e);
+            throw new RuntimeException("Failed to retrieve FlightInfo", e);
+        }
+    }
+
+    @Override
+    public Runnable acceptPut(CallContext callContext, FlightStream flightStream, StreamListener<PutResult> streamListener)
+    {
+        throw new UnsupportedOperationException("This operation is not supported");
+    }
+
+    @Override
+    public void doAction(CallContext callContext, Action action, StreamListener<Result> streamListener)
+    {
+        try {
+            String jsonRequest = new String(action.getBody(), StandardCharsets.UTF_8);
+            JsonNode rootNode = objectMapper.readTree(jsonRequest);
+            String schemaName = rootNode.get("interactionProperties").get("schema_name").asText(null);
+
+            String query;
+            if (schemaName == null) {
+                query = "SELECT SCHEMA_NAME FROM INFORMATION_SCHEMA.SCHEMATA";
+            }
+            else {
+                schemaName = schemaName.toUpperCase();
+                query = "SELECT TABLE_NAME FROM INFORMATION_SCHEMA.TABLES WHERE TABLE_SCHEMA='" + schemaName + "'";
+            }
+            ResultSet rs = connection.createStatement().executeQuery(query);
+            List<String> names = new ArrayList<>();
+            while (rs.next()) {
+                names.add(rs.getString(1));
+            }
+
+            String jsonResponse = objectMapper.writeValueAsString(names);
+            streamListener.onNext(new Result(jsonResponse.getBytes(StandardCharsets.UTF_8)));
+            streamListener.onCompleted();
+        }
+        catch (Exception e) {
+            streamListener.onError(e);
+        }
+    }
```

Comment:
It seems like these are just copied from `TestingArrowServer`, but don't really apply to the test case where they're used.  Can you make it simpler please, seems like this can just be hardcoded since it's not returning TPCH data.

Reply:
Setting hardcoded values.